tests/js/spec/components/search/search.spec.tsx

```diff
@@ -0,0 +1,228 @@
+    ).toBeInTheDocument();
+    expect(
+      screen.queryByText(textWithMarkupMatcher(/Vandelay Industries - Import/))
+    ).not.toBeInTheDocument();
+  });
+
+  it('navigates to a route when item has to prop', () => {
+    mountWithTheme(
+      <Search
+        {...makeSearchProps({
+          sources: [
+            makeSearchResultsMock([
+              {
+                resultType: 'integration',
+                sourceType: 'organization',
+                title: 'Vandelay Industries - Import',
+                to: 'https://vandelayindustries.io/import',
+                model: {slug: 'vdl-imp'},
+              },
+            ]),
+          ],
+        })}
+      />,
+      {
+        context: TestStubs.routerContext(),
+      }
+    );
+
+    const opener = {opener: 'Sentry.io', location: {href: null}};
+
+    // @ts-ignore this is a partial mock of the window object
+    const windowSpy = jest.spyOn(window, 'open').mockReturnValue(opener);
+
+    userEvent.click(screen.getByPlaceholderText('Search Input'));
+    userEvent.keyboard('Import');
+
+    userEvent.click(
+      screen.getByText(textWithMarkupMatcher(/Vandelay Industries - Import/))
+    );
+
+    expect(windowSpy).toHaveBeenCalledTimes(1);
+    expect(opener.opener).not.toBe('Sentry.io');
+    expect(opener.location.href).toBe('https://vandelayindustries.io/import');
+  });
+
+  it('calls item action when it is a function', () => {
+    mountWithTheme(
+      <Search
+        {...makeSearchProps({
+          sources: [
+            makeSearchResultsMock([
+              {
+                resultType: 'integration',
+                sourceType: 'organization',
+                title: 'Vandelay Industries - Import',
+                to: 'https://vandelayindustries.io/import',
+                model: {slug: 'vdl-imp'},
+              },
+            ]),
+          ],
+        })}
+      />,
+      {
+        context: TestStubs.routerContext(),
+      }
+    );
+
+    const opener = {opener: 'Sentry.io', location: {href: null}};
+
+    // @ts-ignore this is a partial mock of the window object
+    const windowSpy = jest.spyOn(window, 'open').mockReturnValue(opener);
+
+    userEvent.click(screen.getByPlaceholderText('Search Input'));
+    userEvent.keyboard('Import');
+
+    userEvent.click(
+      screen.getByText(textWithMarkupMatcher(/Vandelay Industries - Import/))
+    );
+
+    expect(windowSpy).toHaveBeenCalledTimes(1);
+    expect(opener.opener).not.toBe('Sentry.io');
+    expect(opener.location.href).toBe('https://vandelayindustries.io/import');
+  });
+  it('renders max search results', async () => {
+    const results = new Array(10).fill(0).map((_, i) => {
+      const resultItem: ResultItem = {
+        resultType: 'integration',
+        sourceType: 'organization',
+        title: `${i} Vandelay Industries - Import`,
+        to: 'https://vandelayindustries.io/import',
+        model: {slug: 'vdl-imp'},
+      };
+
+      return resultItem;
+    });
```

Comment:
```suggestion
       const results: ResultItem[] = new Array(10).fill(0).map((_, i) => ({
      resultType: 'integration',
      sourceType: 'organization',
      title: `${i} Vandelay Industries - Import`,
      to: 'https://vandelayindustries.io/import',
      model: {slug: 'vdl-imp'},
    }));
```

Reply:
Hmm, I'm not a fan of the shorthand methods too much - I write most of these this way because I'm not familiar with the types, so it's easier for me to declare them as lowest common type and then have the compiler infer the array type. One small benefit of this is that it also makes TS error messages less cryptic, as you dont get the full type resolution stack in the error, you just see the unknown/missing property error.